### ISSUE TITLE
[New Arch][Android] Touchable components stop responding randomly

### ISSUE THREAD

**Issue by peaktan** [facebook/react-native#48387]
### Description

On Android new Arch, touch events always stop responding randomly, and recently we expanded the user base, and more and more people are reporting obstacles. I think many people have raised such issues before, and finally there was a fix that said it fixed such problems. I wonder whether the situation we encountered can be fixed, because we are using version 0.74.1. Due to the large number of users and high upgrade cost, we want to fix such problems on the basis of 0.74.1. Thank you

### Steps to reproduce

The hardest thing about this is that it can't be reliably reproduced.  See description above.

### React Native Version

0.74.1

### Affected Platforms

Runtime - Android

### Areas

Fabric - The New Renderer

### Output of `npx react-native info`

```text
System:
  OS: macOS 13.4
  CPU: (10) arm64 Apple M2 Pro
  Memory: 123.03 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.0.0
    path: /usr/local/bin/node
  Yarn:
    version: 1.22.19
    path: /usr/local/bin/yarn
  npm:
    version: 8.6.0
    path: /usr/local/bin/npm
  Watchman:
    version: 2024.01.22.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.12.1
    path: /Users/01400926/.rbenv/shims/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 22.4
      - iOS 16.4
      - macOS 13.3
      - tvOS 16.4
      - watchOS 9.4
  Android SDK: Not Found
IDEs:
  Android Studio: 2022.3 AI-223.8836.35.2231.10671973
  Xcode:
    version: 14.3/14E222b
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.13
    path: /usr/bin/javac
  Ruby:
    version: 3.2.2
    path: /Users/01400926/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.74.1
    wanted: 0.74.1
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: Not found
  newArchEnabled: false
```

### Stacktrace or Logs

```text
not
```

### Reproducer

not

### Screenshots and Videos

In this example, the user list doesn't respond, but the navigation at the top switches, the scan hover button clicks, and, oddly enough, touch events are available again after the user swipes a few lists.

https://github.com/user-attachments/assets/0f2bfe1e-dd42-48e8-9454-9b7b7d06cb83

**Comment by TallNutAlt** [comment]
#44643 Is it related to this

**Comment by TallNutAlt** [comment]
It may be related to this #44643

**Comment by peaktan** [comment]
> 此事有关#44643

Hello, can this fix fix this problem? I haven't tried it yet. https://github.com/facebook/react-native/pull/45865

**Comment by TallNutAlt** [comment]
> > 此事有关#44643
> 
> Hello, We are not using react-native-pager-view，can this fix fix this problem? I haven't tried it yet. #45865

我并不清楚此pr是否能解决此问题，但是目前此问题仅在新构架被发现

**Comment by peaktan** [comment]
> > > 此事有关#44643
> > 
> > 
> > Hello, We are not using react-native-pager-view，can this fix fix this problem? I haven't tried it yet. #45865
> 
> 我并不清楚此pr是否能解决此问题，但是目前此问题仅在新构架被发现

Yes, especially the huawei mobile phones, can emersion, source of the study showed that the problem seems to be Pressability, js, I try to _isTouchWithinResponderRegion judgment seems to have a problem, lead to trigger occurs, The signal is LEAVE_PRESS_RECT

**Comment by TallNutAlt** [comment]
> > > > 此事有关#44643
> > > 
> > > 
> > > Hello, We are not using react-native-pager-view，can this fix fix this problem? I haven't tried it yet. #45865
> > 
> > 
> > 我并不清楚此pr是否能解决此问题，但是目前此问题仅在新构架被发现
> 
> Yes, especially the huawei mobile phones, can emersion, source of the study showed that the problem seems to be `Pressability.js`, I try to `_isTouchWithinResponderRegion` judgment seems to have a problem, lead to trigger occurs, The signal is `LEAVE_PRESS_RECT`

此pr不能解决此问题的话，你可能需要等待rn安卓团队的人回来了

**Comment by princektripathi** [comment]
@peaktan, can you please share a reproducer to validate? It will also help to narrow down if there is any other dependency that could possibly cause this issue.

**Comment by peaktan** [comment]
> @peaktan, can you please share a reproducer to validate? It will also help to narrow down if there is any other dependency that could possibly cause this issue.

Hello, this demo can be reproduced on Huawei mobile phone, there is no problem with the simulator and some other equipment, only some machines have problems, please help to see the reason, thank you：https://github.com/peaktan/rnnotresponsedemo.git

**Comment by Dallas62** [comment]
Hello, also experienced this issue, specially when the button was placed in a Flatlist with Animated header.
Found a workaround by replacing buttons in that list with RN Gesture Handler pressable.
Hope it will be fix soon to remove this tricks...

**Comment by peaktan** [comment]
> Hello, also experienced this issue, specially when the button was placed in a Flatlist with Animated header. Found a workaround by replacing buttons in that list with RN Gesture Handler pressable. Hope it will be fix soon to remove this tricks... Version 0.76.5 - Android.

Yes, this button in FlatList occasionally fails to respond to user clicks, but can be clicked again after sliding. Is this the case for you as well?

**Comment by Dallas62** [comment]
> > Hello, also experienced this issue, specially when the button was placed in a Flatlist with Animated header. Found a workaround by replacing buttons in that list with RN Gesture Handler pressable. Hope it will be fix soon to remove this tricks... Version 0.76.5 - Android.
> 
> Yes, this button in FlatList occasionally fails to respond to user clicks, but can be clicked again after sliding. Is this the case for you as well?

Was only able to reproduce this issue one time on iOS simulator, but never on a real device (iOS / Android). The issue was reported on Android device (Samsung S10e).
Will experiencing this issue on iOS Simulator, I remember I had to click just above the button to make it works. After a full reinstallation, I wasn't able to reproduce on the simulator...
Just tried the RNGH workaround on impacted button and made the test with the impacted customer, was easier to test than rollback due to custom modules.

**Comment by Dallas62** [comment]
Hi @cipolleschi,

Not sure to understand your point. Yes for the iOS issue on simulator, the bug "disapear".
But the issue appears on Android device (Play Store Release), I will not tell end-users to deinstall the application to make it works... (and I think the impacted user reinstalled the application). Also, I was forced to make a workaround with React Native Gesture Handler.

I cannot provide a reproducer since the issue was only perceptible on end-user devices. I will never be sure to reproduce it correctly.

The React Native version was 0.76.x, new architecture.

Can you give the version of RN where RNTester test case was introduced ?

**Comment by Dallas62** [comment]
The only thing I'm sure, is that the user experienced the issue on RN 0.76.3.
Sad that I can't reproduce on my test Android device, maybe someone will encounter the same issue and be able to give more details.

**Comment by rayson-intrack** [comment]
I'm having this same issue in the 0.74, and was using RN Gesture Handler pressable as a workaround, however the issue was only happening on iOS for me. Updating RN to the latest 0.76.5 seems to fix the problem fox iOS, but now the exact same issue is happening on Android.

**Comment by Dallas62** [comment]
Hi @rayson-intrack, as mentioned by @cortinico, are you able to reproduce it on a real device ? or at least do you have a video ?

**Comment by wvq** [comment]
@rayson-intrack   Same problem, iOS works fine, android didn't  work. version is `react-native@0.76.5`.

When I press, the `onPressIn` event is called.
the problem is `onPressOut` is also called quickly (abount 100ms~150ms), I haven't lifted my finger yet.

So the `onPress` callback need me to lift my finger as quick as possible,  must lifted before `onPressOut` calls.

I think there is some thing wrong in android's  gesture dectech.


```bash
 // no trigger onPress log 
 (NOBRIDGE) LOG  onPressIn at:  1736215926067
 (NOBRIDGE) LOG  onPressOut at:  1736215926216
 (NOBRIDGE) LOG  onPressIn at:  1736215927667
 (NOBRIDGE) LOG  onPressOut at:  1736215927811
 (NOBRIDGE) LOG  onPressIn at:  1736215928583
 (NOBRIDGE) LOG  onPressOut at:  1736215928728

   // click as quick as possible to trigger onPress
 (NOBRIDGE) LOG  onPressIn at:  1736216632304
 (NOBRIDGE) LOG  onPressOut at:  1736216632447
 (NOBRIDGE) LOG  onPressIn at:  1736216633106
 (NOBRIDGE) LOG  onPressOut at:  1736216633248
 (NOBRIDGE) LOG  onPressIn at:  1736216633275
 (NOBRIDGE) LOG  onPress at:  1736216633305
 (NOBRIDGE) LOG  onPressOut at:  1736216633416
 (NOBRIDGE) LOG  onPressIn at:  1736216633432
```

**Comment by cipolleschi** [comment]
@wvq you mentioned that it is happening for you on 0.76.5. Can you create a sample reproducer using this [template](https://github.com/react-native-community/reproducer-react-native)?

**Comment by wvq** [comment]
@cipolleschi 

It's simple to reproduce, follow `react navigation` tutorials, add an button to header right, nothing more.

[repo-to-reproduce-pressIn-issue](https://github.com/wvq/reproduce-react-native-touchable-issue.git)

you can see video:



https://github.com/wvq/reproduce-react-native-touchable-issue/blob/main/example.mp4

**Comment by cipolleschi** [comment]
The reason why we ask to create a reproducer is not because it's hard, but because it saves us some time.
The repo has more than 550 issues. If we have to manually create a reproducer ourselves for each of those, it will take an immense amount of time. We are a few people that looks at the issues (you can count us on a hand), so the request for creating a repro is to distribute the load on multiple people and the community to help us be more effective and to use our time better. If we had a repro for every issue, we can look into and fix more issues than if we have to create the repro ourselves. 
It is a mere matter of time.

**Comment by wvq** [comment]
Oh..., I‘m sorry, I didn't blame it.
English is not my native language, so there may be some misunderstandings. I don't meaning it.
What I saying just a description for what I have done to reproduce, It's also aim to save your time for reading code.

**Comment by wvq** [comment]
@cipolleschi  All I found the problem is  `Pressability._isTouchWithinResponderRegion` never return true when button on navigation bar.

The touch's `pageX/pageY` is correct. but the button's `responderRegion` is weird. 
Tracked to `dom.measure`, but C is beyond my ability.

<img width="1243" alt="Image" src="https://github.com/user-attachments/assets/ffe219e9-8403-49b5-a865-6830e6a1c97f" />

**Comment by RikSchefferAmsterdam** [comment]
In our app we also have quite some of the Android users experience that some of the buttons are unresponsive on RN 0.76 with the new architecture enabled, while it previously was working on lower RN versions with the new architecture disabled. In our testing not all Android devices are affected and not completely all of the time. On some devices the buttons react after pressing them for about 20 times on other devices they do not respond at all.

**Comment by xinha-sh** [comment]
> The reason why we ask to create a reproducer is not because it's hard, but because it saves us some time. The repo has more than 550 issues. If we have to manually create a reproducer ourselves for each of those, it will take an immense amount of time. We are a few people that looks at the issues (you can count us on a hand), so the request for creating a repro is to distribute the load on multiple people and the community to help us be more effective and to use our time better. If we had a repro for every issue, we can look into and fix more issues than if we have to create the repro ourselves. It is a mere matter of time.

The main issue is everything work fine on emulator. But when published to play store, the Pressable components stop responding.

**Comment by cipolleschi** [comment]
Just to share some more details:
- The problem happen
- The problem happens when you press the button, move the finger and then lift the finger. This happens regularly on a real device. On the simulator, we use the mouse which is much more precise and usually we just click (press and lift, without moving) and then does not trigger the bug.
- We already fixed a base case of this problem in all the released versions of React Native
- We are aware that the problem persist when you apply transformations to the parent. When this happens the JS pressability returns wrong values and therefore the listeners are not called.

We are investigating proper solutions, but this is not an easy bug to handle, unfrotunately.

**Comment by just-myself** [comment]
0.74.5 versions of Android phones also have occasional occurrences

**Comment by Qdafengzi** [comment]
My app has a "My page" page that uses FlatList, where clicking, swiping, clicking again quickly causes the entire page to fail.
this happened when opened the new architecture

**Comment by Qdafengzi** [comment]
> * We are aware that the problem persist when you apply transformations to the parent. When this happens the JS pressability returns wrong values and therefore the listeners are not called.

The RN new architecture project developed by our company is dead unless this problem is fixed

**Comment by Qdafengzi** [comment]
> Just to share some more details:
> 
> * The problem happen
> * The problem happens when you press the button, move the finger and then lift the finger. This happens regularly on a real device. On the simulator, we use the mouse which is much more precise and usually we just click (press and lift, without moving) and then does not trigger the bug.
> * We already fixed a base case of this problem in all the released versions of React Native
> * We are aware that the problem persist when you apply transformations to the parent. When this happens the JS pressability returns wrong values and therefore the listeners are not called.
> 
> We are investigating proper solutions, but this is not an easy bug to handle, unfrotunately.

@cipolleschi As you mentioned above, how can I avoid this problem in app? Can you explain in more detail what causes this problem?

**Comment by ktcer** [comment]
> [@cipolleschi](https://github.com/cipolleschi) All I found the problem is `Pressability._isTouchWithinResponderRegion` never return true when button on navigation bar.
> 
> The touch's `pageX/pageY` is correct. but the button's `responderRegion` is weird. Tracked to `dom.measure`, but C is beyond my ability.
> 
> <img alt="Image" width="1243" src="https://private-user-images.githubusercontent.com/14317333/404904993-ffe219e9-8403-49b5-a865-6830e6a1c97f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMzNTgxMDMsIm5iZiI6MTc1MzM1NzgwMywicGF0aCI6Ii8xNDMxNzMzMy80MDQ5MDQ5OTMtZmZlMjE5ZTktODQwMy00OWI1LWE4NjUtNjgzMGU2YTFjOTdmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzI0VDExNTAwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjOWM2NzJhMDg4OWFhOTUyODc0ZWIyODE5NGFmMGFiYWYzYTk5NmNhYjc0YWUxNWFjODQ5NWNlMjgyMzIyZjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.h916YrtuELLl_CkgeKYIh-0oINe57Bxfu4y5mC-69jQ">

How to solve it? Encountering the same problem

**Comment by Qdafengzi** [comment]
update the 0.82.1 problem still exist  😅😅😅

**Comment by antoxa2584x** [comment]
Same here. On one of our main devices, only RectButton from react-native-gesture-handler works

**Comment by AlexCernik** [comment]
Same problem in React Native 0.85.0 and Android 10.

For some reason, the first time I access a StackNavigator screen, it works, but if I edit the parameter with `popTo('Screen', { refreshed: Date.now() });`, the all pressable elements don't work very well, and I have to perform several touches for `onPress` to execute. I also tried this, but the problem persists:

```javascript
dispatch({
...CommonActions.replaceParams({ refreshed: Date.now() }),
source: getState().routes.find(r => r.name === 'Screen').key
});
```

For some reason, if onPressIn is used instead of onPress, this problem apparently does not occur.

**Comment by LiamMorrow** [comment]
I have the same problem, and have for over a year. The only workaround I've found is to wrap all my components to use react-native-gestures. The problem has never occured in components which use this. I'm mainly commenting because I noticed it happening again on components in my app which don't use my wrappers. 

Example here:
https://github.com/LiamMorrow/LiftLog/blob/main/app%2Fsrc%2Fcomponents%2Fpresentation%2Ffoundation%2Ficon-button.tsx
